### DISCUSSION TITLE
[Question] Is there any plan for fluss to be compatible with kafka protocol?

### DISCUSSION THREAD

**Discussion by wking1986** [apache/fluss#94] [Q&A] [answered]
### Search before asking

- [X] I searched in the [issues](https://github.com/alibaba/fluss/issues) and found nothing similar.

### Motivation

Kafka is widely used in the field of streaming storage. Being compatible with the Kafka protocol can facilitate the migration of existing systems and reduce the cost of connecting new systems.

### Willingness to contribute

- [ ] I'm willing to submit a PR!

**Comment by polyzos** [comment, marked as answer]
Yes, kafka compatibility is on the [roadmap](https://alibaba.github.io/fluss-docs/roadmap/)

**Reply by litiliu** [reply]
> Yes, kafka compatibility is on the [roadmap](https://alibaba.github.io/fluss-docs/roadmap/)

Does the kafka compatibility on the [roadmap](https://alibaba.github.io/fluss-docs/roadmap/) also inlcluding kafka read? We have lots of systems using the kafka-client to read kafka message in stream.  Suppoting kafka protocol read makes migrating easier.

**Reply by wuchong** [reply]
Yes. The planned kafka compatibility is wire protocol compatibility including all the common APIs. So existing applications can use Fluss as a drop-in replacement.

**Comment by wuchong** [comment]
The umbrella issue of kafka compatibility: https://github.com/alibaba/fluss/issues/486
